### PR TITLE
in-memory-engine: replace region with CacheRegion 

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #16141

What's Changed:

```commit-message
Previous in #17331, we refactor the RangeManager and use metapb::Region to store region info. As  metapb::Region is very cumbersome, this PR replace it with `CacheRegion`, renaming from `CacheRange` and adding `id` and `epoch_version`. 
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/engine_traits/src/range_cache_engine.rs [hunk 1]
```diff
@@ -105,37 +103,58 @@ pub trait RangeCacheEngineExt {
...
+        id: u64,
+        epoch_version: u64,
+        start: T1,
+        end: T2,
+    ) -> Self {
+        Self {
+            id,
+            epoch_version,
+            start: start.into(),
+            end: end.into(),
```
Comment (overvenus) [inline components/engine_traits/src/range_cache_engine.rs:141]: How about adding asserts `start` and `end` that they start with 'z'?

File: components/engine_traits/src/range_cache_engine.rs [hunk 2]
```diff
@@ -105,37 +103,58 @@ pub trait RangeCacheEngineExt {
...
             end: enc_end_key(region),
+            id: region.id,
+            epoch_version: region.get_region_epoch().version,
         }
     }
 }
 
-impl CacheRange {
+impl CacheRegion {
     // todo: need to consider ""?
```
Comment (overvenus) [inline components/engine_traits/src/range_cache_engine.rs:156]: Stale todo if we assert `start`/`end` keys are data keys.

**Review by SpadeA-Tang — APPROVED** [review]

File: components/engine_traits/src/range_cache_engine.rs (hunk 1 above)
Comment (glorv) [reply components/engine_traits/src/range_cache_engine.rs:141]: `CacheRange::new` is only used for test, seems some old cases uses ranges that are not actually fall in the data range.

**Pull request merged**